### PR TITLE
Added renderer config for WebGL renderer

### PR DESCRIPTION
**Pull request by hotbrainy**
Can we add renderer config in constructor?

### CODE REVIEW

File: packages/core/src/model.ts [hunk 1]
```diff
@@ -355,6 +360,7 @@ export type ParsedViewerConfig = Omit<
     | 'defaultPitch'
     | 'fisheye'
     | 'requestHeaders'
+    | 'renderer'
```
Comment (mistic100) [inline packages/core/src/model.ts:363]: This as no use. This type is for when the result of the parser is different than the input.

File: packages/core/src/services/Renderer.ts [hunk 2]
```diff
@@ -60,7 +60,8 @@ export class Renderer extends AbstractService {
     constructor(viewer: Viewer) {
         super(viewer);
 
-        this.renderer = new WebGLRenderer({ alpha: true, antialias: true });
+        this.renderer = new WebGLRenderer(this.config.renderer);
+//         this.renderer = new WebGLRenderer({ alpha: true, antialias: true });
```
Comment (mistic100) [inline packages/core/src/services/Renderer.ts:64]: To remove

**Review by mistic100 — COMMENTED** [review]
Please update the documentation as well (at the very end of the advanced configuration section)

**Comment by mistic100** [comment]
By the way this flag is not needed to make a screenshot, you only have to get the canvas content immediately after rendering the Three.js scene.

To do that with PSV you can use the render event and force a render :

```js
viewer.addEventListener('render', () => {
    const link = document.createElement('a');
    link.download = 'screenshot.png';
    link.href = viewer.renderer.renderer.domElement.toDataURL();
    link.click();
    setTimeout(() => {
        document.removeChild(link);
    }, 100);
}, { once: true });

viewer.needsUpdate();
```

If that's okay, please close this pull request

**Comment by hotbrainy** [comment]
> Please update the documentation as well (at the very end of the advanced configuration section)

Added the updated option as you mentioned.
thanks in advance.

**Comment by mistic100** [comment]
@hotbrainy did you see my other comment about doing screenshots without this option ?

**Comment by mistic100** [comment]
If there is no real need I don't want to add "useless" options.

**Comment by Amoki** [comment]
I want to use https://github.com/niklasvh/html2canvas with the viewer and `preserveDrawingBuffer` is required. This PR would solve my problem :)

**Comment by Amoki** [comment]
It gives me the error: ` Unable to clone WebGL context as it has preserveDrawingBuffer=false`

**Pull request closed without merging**